### DISCUSSION TITLE
Cannot change WebUI port on bare metal python installation

### DISCUSSION THREAD

**Discussion by i0ntempest** [open-webui/open-webui#7483] [General]
# Bug Report

## Installation Method

pip with venv

## Environment

- **Open WebUI Version:** 0.4.6
- **Ollama (if applicable):** 0.4.6

- **Operating System:** macOS 15.0.2
- **Browser (if applicable):** N/A

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [ ] I have included the Docker container logs.
- [x] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:
WebUI port is configurable per official guide

## Actual Behavior:
It is hardcoded in `__init__.py`

## Description

**Bug Summary:**
Setting env var `PORT` does not change WebUI port

## Reproduction Details

**Steps to Reproduce:**
Follow the [official guide](https://docs.openwebui.com/getting-started/advanced-topics/env-configuration#port) to set the environment variable `PORT=38080` and run `open-webui serve`. The webui port stays on 8080.

## Logs and Screenshots

**Browser Console Logs:**
N/A

**Docker Container Logs:**
N/A

**Screenshots/Screen Recordings (if applicable):**
N/A

**Comment by i0ntempest** [comment]
Is this not a bug or am I missing something? Why is this converted? @tjbck

**Reply by tjbck** [reply]
`PORT` env var exists for docker deployment. `open-webui serve --port 8081` is what you're looking for.

With that being said, I'd appreciate if you could just do a quick search for relevant posts next time.

**Reply by i0ntempest** [reply]
Thanks - that explains it. But the fact that all other env vars listed in the documentation works with bare metal deployment EXCEPT this one is kinda confusing.
